torch/_prims_common/wrappers.py

```diff
@@ -191,8 +191,18 @@ def _out_wrapper(fn: Callable) -> Callable:
             )
         )
 
+        sig = inspect.signature(fn)
+        factory_kwargs = ("device", "dtype")
```

Comment:
This doesn't seem sufficient; it seems layout is also propagated:

```
>>> torch.zeros(3, out=torch.empty(3, dtype=torch.long, layout=torch.sparse_coo))
tensor(indices=tensor([], size=(1, 0)),
       values=tensor([], size=(0,)),
       size=(3,), nnz=0, layout=torch.sparse_coo)
```

Reply:
Yup, this would need to be updated - leaving this as a follow up as most factory functions implemented as refs only support the dtype and device kwargs